### PR TITLE
Modify ERI test for EAM

### PR DESCRIPTION
**Pull request by rljacob**
ERI will edit user_nl_cam to run.  Need to optionally
edit user_nl_eam (if it exists).

Test suite: ERI.ne4_ne4.FC5AV1C-L
Test baseline:  none
Test namelist changes:  none
Test status: BFB

### CODE REVIEW

File: scripts/lib/CIME/SystemTests/eri.py [hunk 1]
```diff
@@ -112,11 +112,15 @@ def run_phase(self):
 
         dout_sr1 = clone1.get_value("DOUT_S_ROOT")
 
-        # force cam namelist to write out initial file at end of run
+        # force cam/eam namelist to write out initial file at end of run
         if os.path.exists("user_nl_cam"):
             if "inithist" not in open("user_nl_cam", "r").read():
                 with open("user_nl_cam", "a") as fd:
                     fd.write("inithist = 'ENDOFRUN'\n")
+        if os.path.exists("user_nl_eam"):
```
Comment (jedwards4b) [inline scripts/lib/CIME/SystemTests/eri.py:120]: how about:
```
  for model in ["cam", "eam"]:
    user_nl = "user_nl_{}".format(model)
    if os.path.exists(user_nl):
      if "inithist" not in open(user_nl, "r").read():
        with open(user_nl, "a") as fd:
          fd.write("inithist = 'ENDOFRUN'\n")
```

**Review by jedwards4b — CHANGES_REQUESTED** [review]

**Comment by rljacob** [comment]
@jedwards4b changed it to your version and force-pushed to remove my other version.

**Comment by rljacob** [comment]
Related issue #3964

**Review by jedwards4b — APPROVED** [review]

**Comment by jedwards4b** [comment]
Pylint issue:
```
======================================================================
FAIL: test_pylint_scripts_lib_CIME_SystemTests_eri_py (__main__.B_CheckCode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 3497, in test
    self.assertTrue(result == "", msg=result)
AssertionError: False is not true : ************* Module CIME.SystemTests.eri
scripts/lib/CIME/SystemTests/eri.py:117:0: W0311: Bad indentation. Found 10 spaces, expected 12 (bad-indentation)
scripts/lib/CIME/SystemTests/eri.py:118:0: W0311: Bad indentation. Found 10 spaces, expected 12 (bad-indentation)
scripts/lib/CIME/SystemTests/eri.py:119:0: W0311: Bad indentation. Found 12 spaces, expected 16 (bad-indentation)
scripts/lib/CIME/SystemTests/eri.py:120:0: W0311: Bad indentation. Found 14 spaces, expected 20 (bad-indentation)
scripts/lib/CIME/SystemTests/eri.py:121:0: W0311: Bad indentation. Found 16 spaces, expected 24 (bad-indentation)
```

**Comment by rljacob** [comment]
Can I run pylint on the command line to check that I have it right?

**Comment by jedwards4b** [comment]
scripts_regression_tests.py runs pylint as one of its tests.  You should have run it as a step in submitting the PR.  But you can also run cime/scripts/Tools/code_checker eri.py

File: scripts/lib/CIME/SystemTests/eri.py [hunk 2]
```diff
@@ -112,11 +112,13 @@ def run_phase(self):
...
         dout_sr1 = clone1.get_value("DOUT_S_ROOT")
 
-        # force cam namelist to write out initial file at end of run
-        if os.path.exists("user_nl_cam"):
-            if "inithist" not in open("user_nl_cam", "r").read():
-                with open("user_nl_cam", "a") as fd:
-                    fd.write("inithist = 'ENDOFRUN'\n")
+        # force cam/eam namelist to write out initial file at end of run
+        for model in ["cam", "eam"]:
+          user_nl = "user_nl_{}".format(model)
```
Comment (billsacks) [inline scripts/lib/CIME/SystemTests/eri.py:117]: Just for informational purposes (doesn't need to be fixed for this PR): this doesn't handle multi-instance cases. There is a more general function here https://github.com/esmci/cime/blob/master/scripts/lib/CIME/SystemTests/test_utils/user_nl_utils.py#L8 that handles multi-instance, but that function would need to be extended if you want to replicate all of the logic here, including something like an optional argument to return quietly if the given file doesn't exist.

**Comment by rljacob** [comment]
pylint issues fixed.  Had to get pylint installed on my dev platform.

**Pull request merged**